site/app/libraries/database/DatabaseQueries.php

```diff
@@ -3331,6 +3331,30 @@ public function getRegistrationSections() {
         return $this->course_db->rows();
     }
 
+    /**
+     * Gets the default registration section from the sections_registration table
+     * @return array<mixed>
+     */
+    public function getDefaultRegistrationSection(): array {
+        $this->course_db->query("SELECT * FROM sections_registration where default_section=true");
+        return $this->course_db->row();
+    }
+
+    /**
+     * Updates the default registration section for self register courses.
+     *
+     */
+    public function setDefaultRegistrationSection(string $section_id): bool {
+        $this->unsetDefaultRegistrationSection();
+        $this->course_db->query("UPDATE sections_registration set default_section = true where sections_registration_id=?", [$section_id]);
+        return true;
+    }
+
+    public function unsetDefaultRegistrationSection(): bool {
+        $this->course_db->query("UPDATE sections_registration SET default_section = false");
+        return true;
+    }
```

Comment:
What happens if two sections somehow become the default section?  Would it be better to change the schema to instead store the ID of the default section (if applicable) in the course table in the master database?

Reply:
I didn't think of that, that would probably be better.